App.tsx

```diff
@@ -22,7 +23,17 @@ Navigation.registerComponent('UserPage', () => props => (
   </ApolloProvider>
 ));
 
-Navigation.registerComponent('AddUserPage', () => AddUserPage);
+Navigation.registerComponent('AddUserPage', () => props => (
+  <ApolloProvider client={apolloClient}>
+    <AddUserPage componentId={props.componentId} />
+  </ApolloProvider>
+));
+
+Navigation.registerComponent('UserDetailsPage', () => props => (
+  <ApolloProvider client={apolloClient}>
+    <UserDetailsPage id={props.id} />
```

Comment:
essas `props` que vem do `registerComponent` contém: 
- as props que vc passa durante a navegação + 
- as props que o `react-native-navigation` injetam no componente.

com isso, vc pode só fazer o spread das props direto na página. ficaria assim 
```suggestion
    <AddUserPage {...props} />
  </ApolloProvider>
));

Navigation.registerComponent('UserDetailsPage', () => props => (
  <ApolloProvider client={apolloClient}>
    <UserDetailsPage {...props} />
```